1_fetch/src/download_nwis_site_data.R

```diff
@@ -0,0 +1,21 @@
+# This function downloads data for a NWIS site and has a built in function to return an error
+# if the download fails.
+download_nwis_site_data <- function(site_num, parameterCd, startDate, endDate, out_dir){
+  
+  # readNWISdata is from the dataRetrieval package
+  data_out <- readNWISdata(sites=site_num, service="iv", 
+                           parameterCd = parameterCd, 
+                           startDate = startDate, 
+                           endDate = endDate)
+
+  # -- simulating a failure-prone web-sevice here, do not edit --
+  set.seed(Sys.time())
+  if (sample(c(T,F,F,F), 1)){
+    stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
+  }
+  # -- end of do-not-edit block
+  
+ readr::write_csv(data_out, file = file.path(out_dir))
```

Comment:
I would rename your argument `out_dir` to `out_file` to make it clear that this contains more than just the directory. Or to be consistent with your other functions, naming this `fileout` might make the most sense.